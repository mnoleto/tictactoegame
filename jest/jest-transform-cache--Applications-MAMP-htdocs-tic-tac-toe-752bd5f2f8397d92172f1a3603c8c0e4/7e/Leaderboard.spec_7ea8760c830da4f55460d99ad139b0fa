/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_Leaderboard = require('./Leaderboard'); /* istanbul ignore next */var _Leaderboard2 = _interopRequireDefault(_Leaderboard);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// Leaderboard is responsible for render the game grid
// CONTRACTS
// 1. 

describe('Leaderboard', function () {
  var props = /* istanbul ignore next */void 0,mountedLeaderboard = /* istanbul ignore next */void 0;
  var leaderboard = function leaderboard() {
    if (!mountedLeaderboard) {
      mountedLeaderboard = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Leaderboard2.default, props));

    }
    return mountedLeaderboard;
  };

  beforeEach(function () {
    props = {};
    mountedLeaderboard = undefined;
  });

  it('always render the Leaderboard', function () {
    var leaderboardElement = leaderboard().find('div');
    expect(leaderboardElement.length).toEqual(1);
  });
});