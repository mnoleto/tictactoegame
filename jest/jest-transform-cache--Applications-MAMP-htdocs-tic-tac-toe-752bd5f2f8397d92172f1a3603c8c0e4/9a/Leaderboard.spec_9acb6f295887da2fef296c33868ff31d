/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_Leaderboard = require('./Leaderboard'); /* istanbul ignore next */var _Leaderboard2 = _interopRequireDefault(_Leaderboard);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// Leaderboard is responsible for render the game grid
// CONTRACTS
// 1. 

describe('Leaderboard', function () {
  var props = /* istanbul ignore next */void 0,mountedLeaderboard = /* istanbul ignore next */void 0;
  var leaderboard = function leaderboard() {
    if (!mountedLeaderboard) {
      mountedLeaderboard = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Leaderboard2.default, props));

    }
    return mountedLeaderboard;
  };

  beforeEach(function () {
    props = {
      allPlayers: [] };

    mountedLeaderboard = undefined;
  });

  it('should render the Leaderboard', function () {
    var leaderboardWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Leaderboard2.default, /* istanbul ignore next */null));
    expect(leaderboardWrapper).toMatchSnapshot();
  });

  it('div contains everything else that gets rendered', function () {
    var leaderboardElement = leaderboard().find('div');
    var wrappingDiv = leaderboardElement.first();
    expect(wrappingDiv.children()).toEqual(leaderboard().children());
  });
});