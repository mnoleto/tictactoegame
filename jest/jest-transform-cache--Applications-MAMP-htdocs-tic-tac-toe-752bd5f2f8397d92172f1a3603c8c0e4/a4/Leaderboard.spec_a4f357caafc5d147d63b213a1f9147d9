/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_Leaderboard = require('./Leaderboard'); /* istanbul ignore next */var _Leaderboard2 = _interopRequireDefault(_Leaderboard);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// Leaderboard is responsible for render the game grid
// CONTRACTS
// 1. 

describe('Leaderboard', function () {
  var props = /* istanbul ignore next */void 0,mountedLeaderboard = /* istanbul ignore next */void 0;
  var leaderboard = function leaderboard() {
    if (!mountedLeaderboard) {
      mountedLeaderboard = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Leaderboard2.default, props));

    }
    return mountedLeaderboard;
  };

  beforeEach(function () {
    props = {
      allPlayers: [] };

    mountedLeaderboard = undefined;
  });

  it('should render the Leaderboard', function () {
    var leaderboardWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Leaderboard2.default, /* istanbul ignore next */null));
    expect(leaderboardWrapper).toMatchSnapshot();
  });

  it('div contains everything else that gets rendered', function () {
    var leaderboardElement = leaderboard().find('div');
    var wrappingDiv = leaderboardElement.first();
    expect(wrappingDiv.children()).toEqual(leaderboard().children());
  });

  it('should have a title h4 with content Leaderboard', function () {
    var titleElement = leaderboard().find('h4');
    expect(titleElement.length).toEqual(1);
    expect(titleElement.text()).toEqual('Leaderboard');
  });


  it('should have a table', function () {
    var tableElement = leaderboard().find('table');
    expect(tableElement.length).toEqual(1);
  });

  it('should have a thead', function () {
    var theadElement = leaderboard().find('thead');
    expect(theadElement.length).toEqual(1);
  });

  it('should have a tr inside thead', function () {
    var theadElement = leaderboard().find('thead');
    var trElement = theadElement.find('tr');
    expect(trElement.length).toEqual(1);
  });

  it('should have 4 th inside thead', function () {
    var theadElement = leaderboard().find('thead');
    var thElements = theadElement.find('th');
    expect(thElements.length).toEqual(4);
  });

  it('should have a tbody', function () {
    var tbodyElement = leaderboard().find('tbody');
    expect(tbodyElement.length).toEqual(1);
  });

  it('should have a 0 trs inside tbody', function () {
    var tbodyElement = leaderboard().find('tbody');
    var trElement = tbodyElement.find('tr');
    expect(trElement.length).toEqual(0);
  });

  describe('allPlayers props filled', function () {
    beforeEach(function () {
      props.allPlayers = [{ name: 'Player 1', wins: 0, defeats: 0, draws: 0 }, { name: 'Player 2', wins: 3, defeats: 0, draws: 0 }];
    });

    it('should have as many tr as allPlayers props length', function () {
      var tbodyElement = leaderboard().find('tbody');
      var trElement = tbodyElement.find('tr');
      expect(trElement.length).toEqual(2);
    });

    it('should order the players by score', function () {
      var tbodyElement = leaderboard().find('tbody');
      var trElement = tbodyElement.childAt(0);
      var tdElement = trElement.childAt(1);
      expect(tdElement.text()).toEqual('Player 2');
    });
  });
});