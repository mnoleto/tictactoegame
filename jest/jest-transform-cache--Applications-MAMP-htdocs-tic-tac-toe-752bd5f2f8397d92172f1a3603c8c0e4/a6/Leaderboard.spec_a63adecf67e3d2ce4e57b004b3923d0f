/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_Leaderboard = require('./Leaderboard'); /* istanbul ignore next */var _Leaderboard2 = _interopRequireDefault(_Leaderboard);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// Leaderboard is responsible for render the game grid
// CONTRACTS
// 1. 

describe('Leaderboard', function () {
  var props = /* istanbul ignore next */void 0,mountedLeaderboard = /* istanbul ignore next */void 0;
  var leaderboard = function leaderboard() {
    if (!mountedLeaderboard) {
      mountedLeaderboard = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Leaderboard2.default, props));

    }
    return mountedLeaderboard;
  };

  beforeEach(function () {
    props = {
      allPlayers: [] };

    mountedLeaderboard = undefined;
  });

  it('should render the Leaderboard', function () {
    var leaderboardWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Leaderboard2.default, /* istanbul ignore next */null));
    expect(leaderboardWrapper).toMatchSnapshot();
  });

  it('div contains everything else that gets rendered', function () {
    var leaderboardElement = leaderboard().find('div');
    var wrappingDiv = leaderboardElement.first();
    expect(wrappingDiv.children()).toEqual(leaderboard().children());
  });

  it('should have a title h4 with content Leaderboard', function () {
    var titleElement = leaderboard().find('h4');
    expect(titleElement.length).toEqual(1);
    expect(titleElement.text()).toEqual('Leaderboard');
  });


  it('should have a table', function () {
    var tableElement = leaderboard().find('table');
    expect(tableElement.length).toEqual(1);
  });

  it('should have a thead', function () {
    var theadElement = leaderboard().find('thead');
    expect(theadElement.length).toEqual(1);
  });

  it('should have a tr inside thead', function () {
    var theadElement = leaderboard().find('thead');
    var trElement = theadElement.find('tr');
    expect(trElement.length).toEqual(1);
  });

  it('should have 4 th inside thead', function () {
    var theadElement = leaderboard().find('thead');
    var thElements = theadElement.find('th');
    expect(thElements.length).toEqual(4);
  });

  it('should have a tbody', function () {

  });

  it('should have a 0 trs inside tbody', function () {

  });

  describe('allPlayers props filled', function () {
    it('should have as many tr as allPlayers props length', function () {

    });

    it('should order the players by score', function () {

    });
  });
});