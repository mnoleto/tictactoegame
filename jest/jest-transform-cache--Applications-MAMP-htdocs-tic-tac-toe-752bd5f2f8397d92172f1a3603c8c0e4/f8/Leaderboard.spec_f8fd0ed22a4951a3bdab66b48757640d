/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_Leaderboard = require('./Leaderboard'); /* istanbul ignore next */var _Leaderboard2 = _interopRequireDefault(_Leaderboard);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// Leaderboard is responsible for render the game grid
// CONTRACTS
// 1. 

describe('Leaderboard', function () {
  var props = /* istanbul ignore next */void 0,mountedLeaderboard = /* istanbul ignore next */void 0;
  var leaderboard = function leaderboard() {
    if (!mountedLeaderboard) {
      mountedLeaderboard = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Leaderboard2.default, props));

    }
    return mountedLeaderboard;
  };

  beforeEach(function () {
    props = {
      allPlayers: [] };

    mountedLeaderboard = undefined;
  });

  it('should render the Leaderboard', function () {
    var leaderboardWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Leaderboard2.default, /* istanbul ignore next */null));
    expect(leaderboardWrapper).toMatchSnapshot();
  });

  it('div contains everything else that gets rendered', function () {
    var leaderboardElement = leaderboard().find('div');
    var wrappingDiv = leaderboardElement.first();
    expect(wrappingDiv.children()).toEqual(leaderboard().children());
  });

  it('should have a title h4', function () {

  });


  it('should have a table', function () {

  });

  it('should have a thead', function () {

  });

  it('should have a tr inside thead', function () {

  });

  it('should have 4 th inside thead', function () {

  });

  it('should have a tbody', function () {

  });

  it('should have a 0 trs inside tbody', function () {

  });

  describe('allPlayers props filled', function () {
    it('should have as many tr as allPlayers props length', function () {

    });

    it('should have as 5 td to each tr', function () {

    });

    it('should the first td has the player name', function () {

    });

    it('should the second td has the wins score', function () {

    });

    it('should the third td has the defeats score', function () {

    });

    it('should the third td has the draws score', function () {

    });

    it('should order the players by score', function () {

    });
  });
});