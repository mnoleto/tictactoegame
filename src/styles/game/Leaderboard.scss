@import '../_colors.scss';

.leaderboard {
  color: $dark-blue;
  display: block;
  margin: 0 auto 30px;
  width: 100%;
  &:after {
    clear: both;
    content: "";
    display: block;
  }
  .title {
    margin-bottom: 5px;
    text-align: center;
  }
  table {
    background: $white;
    margin: 0 auto;
    width: 50%;
  }
  td, th {
    padding: 3px 10px;
    text-align: center;
  }
  tr.heading {
    background-color: lighten($blue, 30%);
  }
  td.name, th.name {
    text-align: left;
  }
  td.position {
    font-weight: 700;
    width: 30px;
  }
}

@media (min-width: 768px) {
  .leaderboard {
    width: 740px;
    table {
      width: 100%;
    }
  }
}

@media (min-width: 992px) {
  .leaderboard {
    width: 960px;
    table {
      width: 70%;
    }
  }
}

@media (min-width: 1200px) {
  .leaderboard {
    width: 1170px;
    table {
      width: 50%;
    }
  }
}