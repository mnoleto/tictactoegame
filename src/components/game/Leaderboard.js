import React, { Component, PropTypes } from 'react';
import '../../styles/game/Leaderboard.scss';

class Leaderboard extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { allPlayers } = this.props;

    let playersItens;
    if(allPlayers && allPlayers.length > 0) {
      const orderedPlayers = allPlayers.sort((a, b) => {
        let scoreA = (parseInt(a.wins)*3) + parseInt(a.draws),
          scoreB = (parseInt(b.wins)*3) + parseInt(b.draws);
        
        return scoreB - scoreA;
      });

      playersItens = orderedPlayers.map((value, index) => {
        return (
          <tr key={'leaderboard_' + index}>
            <td className="position">{index + 1}</td>
            <td className="name">{value.name}</td>
            <td className="wins">{value.wins}</td>
            <td className="defeats">{value.defeats}</td>
            <td className="draws">{value.draws}</td>
          </tr>
        );
      });
    }

    return (
      <div className="leaderboard">
        <h4 className="title">Leaderboard</h4>
        <table cellPadding="0" cellSpacing="0">
          <thead>
            <tr className="heading">
              <th colSpan="2" className="name">Name</th>
              <th className="wins">Wins</th>
              <th className="defeats">Defeats</th>
              <th className="draws">Draws</th>
            </tr>
          </thead>
          <tbody>
            {playersItens}
          </tbody>
        </table>
      </div>
    );
  }
};

Leaderboard.propTypes = {
  allPlayers: PropTypes.array
};

export default Leaderboard;