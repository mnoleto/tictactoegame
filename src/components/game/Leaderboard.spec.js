import React from 'react';
import { mount, shallow } from 'enzyme';
import Leaderboard from './Leaderboard';

// Leaderboard is responsible for render the players leaderboard
describe('<Leaderboard />', () => {
  let props, mountedLeaderboard;
  const leaderboard = () => {
    if(!mountedLeaderboard) {
      mountedLeaderboard = mount(
        <Leaderboard {...props} />
      );
    }
    return mountedLeaderboard;
  };

  beforeEach(() => {
    props = {
      allPlayers: []
    };
    mountedLeaderboard = undefined;
  });

  it('should render the Leaderboard', () => {
    const leaderboardWrapper = shallow(<Leaderboard />);
    expect(leaderboardWrapper).toMatchSnapshot();
  });

  it('div contains everything else that gets rendered', () => {
    const leaderboardElement = leaderboard().find('div');
    const wrappingDiv = leaderboardElement.first();
    expect(wrappingDiv.hasClass('leaderboard')).toBeTruthy();
    expect(wrappingDiv.children()).toEqual(leaderboard().children());
  });

  it('should have a title h4 with content Leaderboard', () => {
    const titleElement = leaderboard().find('h4');
    expect(titleElement.length).toEqual(1);
    expect(titleElement.text()).toEqual('Leaderboard');
  });


  it('should have a table', () => {
    const tableElement = leaderboard().find('table');
    expect(tableElement.length).toEqual(1);
  });

  it('should have a thead', () => {
    const theadElement = leaderboard().find('thead');
    expect(theadElement.length).toEqual(1);
  });

  it('should have a tr inside thead', () => {
    const theadElement = leaderboard().find('thead');
    const trElement = theadElement.find('tr');
    expect(trElement.length).toEqual(1);
  });

  it('should have 4 th inside thead', () => {
    const theadElement = leaderboard().find('thead');
    const thElements = theadElement.find('th');
    expect(thElements.length).toEqual(4);
  });

  it('should have a tbody', () => {
    const tbodyElement = leaderboard().find('tbody');
    expect(tbodyElement.length).toEqual(1);
  });

  it('should have a 0 trs inside tbody', () => {
    const tbodyElement = leaderboard().find('tbody');
    const trElement = tbodyElement.find('tr');
    expect(trElement.length).toEqual(0);
  });

  describe('allPlayers props filled', () => {
    beforeEach(() => {
      props.allPlayers = [{name: 'Player 1', wins: 0, defeats: 0, draws: 0}, {name: 'Player 2', wins: 3, defeats: 0, draws: 0}];
    });

    it('should have as many tr as allPlayers props length', () => {
      const tbodyElement = leaderboard().find('tbody');
      const trElement = tbodyElement.find('tr');
      expect(trElement.length).toEqual(2);
    });

    it('should order the players by score', () => {
      const tbodyElement = leaderboard().find('tbody');
      const trElement = tbodyElement.childAt(0);
      const tdElement = trElement.childAt(1);
      expect(tdElement.text()).toEqual('Player 2');
    });
  });
});