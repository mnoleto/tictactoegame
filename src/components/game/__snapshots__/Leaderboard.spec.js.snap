exports[`Leaderboard render the Leaderboard 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="leaderboard">
    <h4
        className="title">
        Leaderboard
    </h4>
    <table
        cellPadding="0"
        cellSpacing="0">
        <thead>
            <tr
                className="heading">
                <th
                    className="name"
                    colSpan="2">
                    Name
                </th>
                <th
                    className="wins">
                    Wins
                </th>
                <th
                    className="defeats">
                    Defeats
                </th>
                <th
                    className="draws">
                    Draws
                </th>
            </tr>
        </thead>
        <tbody />
    </table>
</div>,
  "nodes": Array [
    <div
      className="leaderboard">
      <h4
            className="title">
            Leaderboard
      </h4>
      <table
            cellPadding="0"
            cellSpacing="0">
            <thead>
                  <tr
                        className="heading">
                        <th
                              className="name"
                              colSpan="2">
                              Name
                        </th>
                        <th
                              className="wins">
                              Wins
                        </th>
                        <th
                              className="defeats">
                              Defeats
                        </th>
                        <th
                              className="draws">
                              Draws
                        </th>
                  </tr>
            </thead>
            <tbody />
      </table>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Leaderboard />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Leaderboard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="leaderboard">
          <h4
                    className="title">
                    Leaderboard
          </h4>
          <table
                    cellPadding="0"
                    cellSpacing="0">
                    <thead>
                              <tr
                                        className="heading">
                                        <th
                                                  className="name"
                                                  colSpan="2">
                                                  Name
                                        </th>
                                        <th
                                                  className="wins">
                                                  Wins
                                        </th>
                                        <th
                                                  className="defeats">
                                                  Defeats
                                        </th>
                                        <th
                                                  className="draws">
                                                  Draws
                                        </th>
                              </tr>
                    </thead>
                    <tbody />
          </table>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="leaderboard">
          <h4
                    className="title">
                    Leaderboard
          </h4>
          <table
                    cellPadding="0"
                    cellSpacing="0">
                    <thead>
                              <tr
                                        className="heading">
                                        <th
                                                  className="name"
                                                  colSpan="2">
                                                  Name
                                        </th>
                                        <th
                                                  className="wins">
                                                  Wins
                                        </th>
                                        <th
                                                  className="defeats">
                                                  Defeats
                                        </th>
                                        <th
                                                  className="draws">
                                                  Draws
                                        </th>
                              </tr>
                    </thead>
                    <tbody />
          </table>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Leaderboard />,
}
`;
